sasmodels/models/micromagnetic_FF_3D.py

```diff
@@ -0,0 +1,194 @@
+r"""
+Definition
+----------
+This model is a micromagnetic approach to analyse the SANS that arises from
+nanoscale variations in the magnitude and orientation of the magnetization in
+bulk ferromagnets in the approach to magnetic saturation (single domain state).
+Typical materials are cold-worked elemental magnets, hard and soft magnetic 
+nanocomposites, amorphous alloys and precipitates in magnetic steel [#Michels2014]_.
+The magnetic SANS depends on the magnetic interactions, the magnetic microstructure
+(defect/particle size, magnetocrystalline anisotropy, saturation magnetisation)
+and on the applied magnetic field. As shown in [#Michels2016]_ near magnetic 
+saturation the scattering cross-section can be evaluated by means of micromagnetic theory
+
+.. math::
+    I(mathbf{Q}) = I_nuc +I_res+ I_mag(mathbf{Q},H),
+
+with the field-independent nuclear and magnetic residual SANS cross section (due
+to nanoscale spatial variations of the saturation magnetisation) measured at
+complete magnetic saturation. Commonly, a measurement at a high magnetic field
+taken as reference and subtracted from the scattering cross-section at lower
+field to obtain the so-called spin-misalignment scattering cross-section 
+
+.. math::
+    I_mag(mathbf{Q},H)= S_K(Q) R_K(mathbf{Q}, H_i) + S_M(Q) R_M(mathbf{Q}, H_i),
+
+with $H_i$ the internal field, i.e. the external magnetic field corrected for
+demagnetizing effects and the influence of the magnetodipolar field and of the
+magnetic anisotropy [#Bick2013]_. This purely magnetic scattering reflects the
+sensitive response of the transversal magnetisation components with to an
+externally applied magnetic field with a contribution $S_K 	imes R_K$ due to
+perturbations around magnetic anisotropy fields and a term $S_M 	imes R_M$
+related to magnetostatic fields. The alignment of the magnetic moments along
+the magnetic field is disturbed by perturbations in the microstructure. The
+anisotropy-field function $S_K$ depends on the Fourier transform of the magnetic
+anisotropy distribution (strength and orientation) in the material and the
+scattering function of the longitudinal magnetisation $S_M$ reflects the
+variations of the saturation magnetisation, e.g. jumps at the particle-matrix
+interface. $R_K$ and $R_M$ denote the micromagnetic response functions that
+describe the magnetisation distribution around a perturbation in magnetic
+anisotropy and flucutations in the saturation magnetisation value.
+
+.. figure:: img/micromagnetic_FF.png
+
+    Magnetisation distribution around (left) a particle with magnetic easy axis
```

Comment:
you are referring to an image which is good. However no image has been uploaded in this PR causing the CI to fail. FYI the path full path to img/ (the line above is fine) for where you need to place the png figure is sasmodels/sasmodels/models/img.

Reply:
Of course, I'll upload it with the next commit